src/lib/theme/utilities.ts

```diff
@@ -11,5 +12,26 @@ export const rgb = (hexColor: `#${string}`): RGB => {
   return `${r} ${g} ${b}`;
 };
 
-export const css = (variable: keyof typeof variables) =>
-  `rgb(var(${variable}))`;
+export const css = (variable: Variable) => `rgb(var(${variable}))`;
+
+export const toColor = (name: ColorName): RGB => {
+  const [paletteColor, shade] = name.split('.') as [PaletteColor, Shade];
+  const color = colors[paletteColor];
+  if (isHexColor(color)) return rgb(color);
+  if (isPaletteColor(paletteColor)) {
+    const color = palette[paletteColor];
+    if (isShade(shade)) return rgb(color[shade]);
+    return rgb(color.DEFAULT);
+  }
+};
+
+export const isHexColor = (color: unknown): color is HexColor => {
+  if (typeof color !== 'string') return false;
+  return /^#[0-9A-F]{6}$/i.test(color);
+};
+
+export const isPaletteColor = (color: string): color is PaletteColor =>
+  color in palette;
+
+export const isShade = (shade: unknown): shade is Shade =>
+  typeof shade === 'number' || shade === 'DEFAULT';
```

Comment:
Since we're using `name.split('.')` on a string in `toColor` we likely want to account for a `string` here
```suggestion
  typeof shade === 'number' || typeof Number(shade) === 'number' || shade === 'DEFAULT';
```